### PR TITLE
nixos/prometheus-nats-exporter: new module

### PR DESCRIPTION
**Pull request by c0deaddict**
## Description of changes

Package for nats-exporter was already in nixpkgs, but a module was missing. Tested on my local home server.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/monitoring/prometheus/exporters/nats.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+with lib;
+
+let
+
+  cfg = config.services.prometheus.exporters.nats;
+
+in
+{
+  port = 7777;
```
Comment (lblasc) [inline nixos/modules/services/monitoring/prometheus/exporters/nats.nix:11]: Most likely `port` is leftover from testing.

Comment (lblasc) [reply nixos/modules/services/monitoring/prometheus/exporters/nats.nix:11]: ignore, my mistake.

**Review by lblasc — APPROVED** [review]
Works as expected.

```
     networking.firewall.allowedTCPPorts = [
       config.services.prometheus.exporters.nats.port
     ];

    services.prometheus = {
      exporters = {
        nats = {
          enable = true;
          extraFlags = [ "-varz" "-jsz=all" ];
        };
      };
    };
```

**Comment by c0deaddict** [comment]
> Works as expected.
> 
> ```
>      networking.firewall.allowedTCPPorts = [
>        config.services.prometheus.exporters.nats.port
>      ];
> 
>     services.prometheus = {
>       exporters = {
>         nats = {
>           enable = true;
>           extraFlags = [ "-varz" "-jsz=all" ];
>         };
>       };
>     };
> ```

Thanks for reviewing!

File: nixos/modules/services/monitoring/prometheus/exporters/nats.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{ config, lib, pkgs, options }:
```
Comment (aanderse) [inline nixos/modules/services/monitoring/prometheus/exporters/nats.nix:1]:
```suggestion
{ config, lib, pkgs, options, ... }:
```

modules need to leave room for unspecified options as you never know when new ones could be added

**Review by aanderse — CHANGES_REQUESTED** [review]
@c0deaddict are you interested in pushing this PR through to merge?

**Comment by c0deaddict** [comment]
@aanderse Yes :) I've added your suggestion to the commits.

**Comment by aanderse** [comment]
@c0deaddict awesome! are you able to resolve conflicts? and then we're good to merge! thank you :bowing_man:

**Comment by c0deaddict** [comment]
@aanderse should be good now!

**Comment by aanderse** [comment]
sorry to be a hassle but can you please drop the merge commit

**Comment by c0deaddict** [comment]
Ah, yes no problem. I used the github online editor, i see it made a mess of the commits :') Recreated the branch with the changes and force pushed. Should be good now.

**Review by aanderse — APPROVED** [review]
thank you very much!

**Pull request merged**